lucene/core/src/java/org/apache/lucene/util/automaton/RegExp.java

```diff
@@ -439,12 +441,19 @@
   /**
    * Syntax flag, enables all optional regexp syntax.
    */
-  public static final int ALL = 0xffff;
-  
+  public static final int ALL = 0xff;
+      
   /**
    * Syntax flag, enables no optional regexp syntax.
    */
   public static final int NONE = 0x0000;
+  
+  //-----  Matching flags ( > 0xff )  ------
+  
+  /**
+   * Allows case insensitive matching of ASCII characters.
+   */
+  public static final int ASCII_CASE_INSENSITIVE = 0x0100;    
```

Comment:
I thought it might be useful if the flag name reflected the current limitations?

Reply:
But then what would be the other flag to allow unicode support ?

Reply:
UNICODE_CASE_INSENSITIVE or just CASE_INSENSITIVE?
Either way it would cover ASCII and all other UNICODE characters. 
Admittedly slightly odd that the two flags overlap but the alternative is people may assume that a non-qualified name like "CASE_INSENSITIVE" would cover all the bases when we only currently support ASCII.

Reply:
ok fine by me.

Reply:
should we ensure that the flag is not used in the syntax_flags since we merge the two internally ?

Reply:
My assumption was this class was lenient to syntax flags > 0xff before this change so should remain so for BWC reasons